pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -313,3 +313,17 @@ func appendPDBTag(tags []string, pdb sql.NullString) []string {
 	}
 	return append(tags, "pdb:"+pdb.String)
 }
+
+func selectWrapper[T any](c *Check, s T, sql string) error {
+	err := c.db.Select(s, sql)
+	if err != nil && (strings.Contains(err.Error(), "ORA-01012") || strings.Contains(err.Error(), "database is closed")) {
```

Comment:
I think this could be a little better since go-ora has its own error type. I haven't tried this but I think this should work: 

```suggestion
    var oraErr network.OracleError
	if err != nil && errors.As(err, &oraError) && oraError.ErrCode == 1012 {
```

Reply:
That unfortunately doesn't work, because `network.OracleError.ErrCode` isn't initialised for `ORA-1012`. 

Also `network.OracleError`doesn't implement `error`:

```
panic: errors: *target must be interface or implement error

goroutine 249 [running]:
errors.As({0x103750180, 0xc0003621f8}, {0x102a3ed80, 0xc000362210?})
    /usr/local/Cellar/go/1.19.5/libexec/src/errors/wrap.go:89 +0x3df
github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm.selectWrapper[...](0xc0012add40?, 0x102eed4e3, {0x102fddda6?, 0xc000098c58?})
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/oracle.go:322 +0x9e
github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm.(*Check).SysMetrics(0xc0012add40)
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/sysmetrics.go:52 +0x792
github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm.(*Check).Run(0xc0012add40)
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/oracle.go:90 +0x85
github.com/DataDog/datadog-agent/pkg/collector/internal/middleware.(*CheckWrapper).Run(0xc0007424e0?)
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/internal/middleware/check_wrapper.go:42 +0xc4
github.com/DataDog/datadog-agent/pkg/collector/worker.(*Worker).Run(0xc0005ee0a0)
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/worker/worker.go:138 +0x364
github.com/DataDog/datadog-agent/pkg/collector/runner.(*Runner).newWorker.func1()
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/runner/runner.go:128 +0x6d
created by github.com/DataDog/datadog-agent/pkg/collector/runner.(*Runner).newWorker
    /Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/runner/runner.go:125 +0x1cd
```

Reply:
Oops. What about this? (`*OracleError` [implements error](https://github.com/sijms/go-ora/blob/41ca433ee475e64869448c27e6e20ad038613ae4/v2/network/oracle_error.go#L10-L15))
```suggestion
        var oraErr *network.OracleError
	if err != nil && errors.As(err, &oraError) && oraError.ErrCode == 1012 {
```

Reply:
That doesn't work either because `oraError` isn't initialised on `ORA-01012`.